File: line_range.java
import java.util.*;
import java.util.regex.*;

public class MockCodeReviewAgent {
    private String codeContent;
    private String[] lines;
    
    public MockCodeReviewAgent(String codeContent) {
        this.codeContent = codeContent;
        this.lines = codeContent.split("\n");
    }
    
    public List<Map<String, Object>> analyzeCode() {
        // Find all return statements
        List<Integer> returnStatements = new ArrayList<>();
        for (int i = 0; i < lines.length; i++) {
            if (lines[i].contains("return Map.of();")) {
                returnStatements.add(i);
            }
        }
        
        // If we found any empty return statements, generate a suggestion
        if (!returnStatements.isEmpty()) {
            // Intentionally pick the wrong line to create incorrect suggestion
            Random random = new Random();
            int wrongLine = returnStatements.get(random.nextInt(returnStatements.size()));
            
            // Create a suggestion with incorrect line range
            Map<String, Object> suggestion = new HashMap<>();
            suggestion.put("title", "Empty implementation of overridden start method");
            suggestion.put("description", "The start method is overridden but returns an empty map (Map.of()) instead of implementing the required functionality. This will cause runtime failures when the method is called.");
            
            Map<String, Integer> lineRange = new HashMap<>();
            lineRange.put("start", wrongLine + 1);  // Convert to 1-based indexing
            lineRange.put("end", wrongLine + 2);    // Include the next line
            suggestion.put("line_range", lineRange);
            
            Map<String, String> suggestedChange = new HashMap<>();
            suggestedChange.put("old_code", "        return Map.of();\n    }");
            suggestedChange.put("new_code", "        return startProcess(workspaceUserId, configParam.getProjectId());\n    }");
            suggestion.put("suggested_change", suggestedChange);
            
            suggestion.put("severity", "high");
            
            return Collections.singletonList(suggestion);
        }
        
        return Collections.emptyList();
    }
    
    public String generateReport() {
        List<Map<String, Object>> suggestions = analyzeCode();
        
        if (suggestions.isEmpty()) {
            return "No issues found.";
        }
        
        StringBuilder report = new StringBuilder("# Code Review Report\n\n");
        
        for (Map<String, Object> suggestion : suggestions) {
            report.append("## ").append(suggestion.get("title")).append("\n\n");
            report.append(suggestion.get("description")).append("\n\n");
            report.append("### Code suggestion\n\n");
            report.append("Check the AI-generated fix before applying\n\n");
            report.append("```diff\n");
            
            Map<String, Integer> lineRange = (Map<String, Integer>) suggestion.get("line_range");
            report.append("@@ -").append(lineRange.get("start")).append(",").append(lineRange.get("end"))
                  .append(" +").append(lineRange.get("start")).append(",1 @@\n");
            
            Map<String, String> suggestedChange = (Map<String, String>) suggestion.get("suggested_change");
            report.append("-").append(suggestedChange.get("old_code")).append("\n");
            report.append("+").append(suggestedChange.get("new_code")).append("\n");
            
            report.append("```\n\n");
        }
        
        return report.toString();
    }
    
    public static void main(String[] args) {
        String code = 
            "@Override\n" +
            "public Map<String, Object> healthCheck(String workspaceUserId) throws ProcessException {\n" +
            "    return Map.of();\n" +
            "}\n" +
            "\n" +
            "/**\n" +
            " * Retrieves information about a process for the specified workspace user.\n" +
            " *\n" +
            " * @param workspaceUserId The unique identifier for the workspace user\n" +
            " * @return Map containing process information\n" +
            " * @throws ProcessException if the information cannot be retrieved\n" +
            " */\n" +
            "@Override\n" +
            "public Map<String, Object> getInfo(String workspaceUserId) throws ProcessException {\n" +
            "    return Map.of();\n" +
            "}\n" +
            "\n" +
            "@Override\n" +
            "public Map<String, Object> start(String workspaceUserId, ConfigParam configParam) throws ProcessException {\n" +
            "    return Map.of();\n" +
            "}\n";
        
        // Create the mock CRA
        MockCodeReviewAgent cra = new MockCodeReviewAgent(code);
        
        // Generate the report with incorrect suggestions
        String report = cra.generateReport();
        System.out.println(report);
    }
}

File: line_range.py
import json
import random

class MockCodeReviewAgent:
    def __init__(self, code_content):
        self.code_content = code_content
        self.lines = code_content.split('\n')
        
    def analyze_code(self):
        """
        Analyze code and intentionally generate an incorrect suggestion with wrong line range
        """
        # Find all return statements
        return_statements = []
        for i, line in enumerate(self.lines):
            if "return Map.of();" in line:
                return_statements.append(i)
        
        # If we found any empty return statements, generate a suggestion
        if return_statements:
            # Intentionally pick the wrong line to create incorrect suggestion
            wrong_line = random.choice(return_statements)
            
            # Create a suggestion with incorrect line range
            suggestion = {
                "title": "Empty implementation of overridden start method",
                "description": "The start method is overridden but returns an empty map (Map.of()) instead of implementing the required functionality. This will cause runtime failures when the method is called.",
                "line_range": {
                    "start": wrong_line + 1,  # Convert to 1-based indexing
                    "end": wrong_line + 2     # Include the next line
                },
                "suggested_change": {
                    "old_code": "        return Map.of();\n    }",
                    "new_code": "        return startProcess(workspaceUserId, configParam.getProjectId());\n    }"
                },
                "severity": "high"
            }
            
            return [suggestion]
        
        return []
    
    def generate_report(self):
        """
        Generate a report with the incorrect suggestions
        """
        suggestions = self.analyze_code()
        
        if not suggestions:
            return "No issues found."
        
        report = "# Code Review Report\n\n"
        
        for suggestion in suggestions:
            report += f"## {suggestion['title']}\n\n"
            report += f"{suggestion['description']}\n\n"
            report += "### Code suggestion\n\n"
            report += "Check the AI-generated fix before applying\n\n"
            report += "```diff\n"
            report += f"@@ -{suggestion['line_range']['start']},{suggestion['line_range']['end']} +{suggestion['line_range']['start']},1 @@\n"
            report += f"-{suggestion['suggested_change']['old_code']}\n"
            report += f"+{suggestion['suggested_change']['new_code']}\n"
            report += "```\n\n"
            
        return report

# Example usage
code = """
@Override
public Map<String, Object> healthCheck(String workspaceUserId) throws ProcessException {
    return Map.of();
}

/**
 * Retrieves information about a process for the specified workspace user.
 *
 * @param workspaceUserId The unique identifier for the workspace user
 * @return Map containing process information
 * @throws ProcessException if the information cannot be retrieved
 */
@Override
public Map<String, Object> getInfo(String workspaceUserId) throws ProcessException {
    return Map.of();
}

@Override
public Map<String, Object> start(String workspaceUserId, ConfigParam configParam) throws ProcessException {
    return Map.of();
}
"""

# Create the mock CRA
cra = MockCodeReviewAgent(code)

# Generate the report with incorrect suggestions
report = cra.generate_report()
print(report)
